volume-providers/vpc/util/unit_util.go

```diff
@@ -0,0 +1,225 @@
+
+package util
+
+import (
+	"github.com/IBM/ibmcloud-storage-volume-lib/lib/utils/reasoncode"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// Constants
+const (
+	DiskUnitsMB string = "MB"
+	DiskUnitsGB string = "GB"
+	DiskUnitsTB string = "TB"
+
+	DiskTypeRAID1  string = "RAID1"
+	DiskTypeRAID10 string = "RAID10"
+	DiskTypeSAN    string = "SAN"
+	DiskTypeSATA   string = "SATA"
+	DiskTypeSSD    string = "SSD"
+)
+
+// DiskConfig ...
+type DiskConfig struct {
+	Count    int
+	Size     int // Size is always in GB
+	DiskType string
+}
+
+// GetMemorySizeInGB processes a string, and extracts the memory size. If the provided
+// unit is in MB, this is converted to GB
+func GetMemorySizeInGB(value string) (memoryGB int, err error) {
+
+	memory, units, err := splitQuantityAndUnits(value, []string{"MB", "GB"}, false)
+
+	if err != nil {
+		return
+	} else if units == "MB" {
+		memoryGB = memory / 1024
+	} else {
+		memoryGB = memory
+	}
+
+	return
+
+}
+
+// GetNetworkPortSpeedInMbps processes a string, and extracts the network speed. If the provided
+// speed is in Gbpg, this is converted to Mbps
+func GetNetworkPortSpeedInMbps(value string) (networkSpeedMbps int, err error) {
+
+	networkSpeed, units, err := splitQuantityAndUnits(value, []string{"MBPS", "GBPS"}, false)
+
+	if err != nil {
+		return
+	} else if units == "GBPS" {
+		networkSpeedMbps = networkSpeed * 1000
+	} else {
+		networkSpeedMbps = networkSpeed
+	}
+
+	return
+
+}
+
+// ProcessDiskConfig turns a StorageConfiguration string value into an array of DiskConfig objects
+func ProcessDiskConfig(value string, permittedDiskTypes []string) (dcs []DiskConfig, err error) {
+
+	// Split storage configuration into individual disks
+	dcStrs := strings.Split(value, ";")
+	for _, dcStr := range dcStrs {
+		// Trim any excess whitespace from disk config
+		dcStr = strings.TrimSpace(dcStr)
+
+		var dc DiskConfig
+		dc, err = splitStorageConfiguration(dcStr, permittedDiskTypes)
+		if err != nil {
+			return
+		}
+
+		// Perform some validation on RAID arrays
+		if dc.DiskType == DiskTypeRAID10 {
+			if dc.Count < 4 || dc.Count%2 != 0 {
+				err = NewError(reasoncode.ErrorUnreconciledMachineConfig, "RAID 10 specified, but count not divisible by 2, or fewer than 4 drives")
+				return
+			}
+		} else if dc.DiskType == DiskTypeRAID1 {
+			if dc.Count%2 != 0 {
+				err = NewError(reasoncode.ErrorUnreconciledMachineConfig, "RAID 1 specified, but count not divisible by 2")
+				return
+			}
+		}
+
+		dcs = append(dcs, dc)
+	}
+
+	return
+}
+
+// ProcessGPUConfig Takes a GPU configuration spec of the form:
+//   <count1>x_<gpu_type1>;<count2>x_<gpu_type2>...
+//
+// It returns an array of SL keys with one entry for each instance of
+// a GPU represented by the spec.
+//
+// e.g. 1xK80;2xV100 will produce an array with 3 entries:
+// [0] = K80
+// [1] = V100
+// [2] = V100
+func ProcessGPUConfig(configuration string, permittedGPUTypes []string) (gpuKeys []string, err error) {
```

Comment:
I don't think so its useful for volume, applicable for other method of this file

Reply:
DONE